Ui/Component/Listing/Column/TrackAndTrace.php

```diff
@@ -43,26 +45,40 @@ public function prepareDataSource(array $dataSource): array
                 continue;
             }
 
-            if (count($addressParts) === 4) {
-                [$company, $street, $city, $postalCode] = $addressParts;
-            } else {
-                [$street, $city, $postalCode] = $addressParts;
-            }
+            $postalCode = array_slice($addressParts, -1)[self::KEY_POSTCODE];
 
             // Stop if either the barcode or postal code is missing.
             if (! $item['track_number'] || $item['track_number'] === self::VALUE_EMPTY || ! $postalCode) {
                 continue;
             }
 
             $trackNumber = $item['track_number'];
-            $data = $this->getData('name');
+            $countryId   = $this->getCountryWithEntityId($item);
+            $data        = $this->getData('name');
 
             // Render the T&T as a link and add the script to remove the click handler.
-            $trackTrace = (new TrackTraceUrl())->create($trackNumber, $postalCode);
+            $trackTrace  = (new TrackTraceUrl())->create($trackNumber, $postalCode, $countryId);
             $item[$data] = "<a class=\"myparcel-barcode-link\" target=\"_blank\" href=\"$trackTrace\">$trackNumber</a>";
             $item[$data] .= self::SCRIPT_UNBIND_CLICK;
         }
 
         return $dataSource;
     }
+
+    /**
+     * @param array $orderData
+     *
+     * @return mixed
+     */
+    public function getCountryWithEntityId(array $orderData): string
+    {
+        $order     = (ObjectManager::getInstance())->create(Order::class)->load($orderData['entity_id']);
+        $countryId = $order->getShippingAddress()->getCountryId();
+
+        if (! $countryId) {
+            $countryId = AbstractConsignment::CC_NL;
+        }
+
+        return $countryId;
```

Comment:
Wat kan `getCountryId()` returnen? Ik zie liever een `?:` of `??` met die fallback value, leest makkelijker als het bij elkaar staat in de return statement